src/Symfony/Component/HttpFoundation/StreamedJsonResponse.php

```diff
@@ -71,10 +68,24 @@ public function __construct(
 
     private function stream(): void
     {
-        $keys = array_keys($this->generators);
+        $generators = [];
+        $structure = $this->data;
+
+        \array_walk_recursive($structure, function (&$item, $key)  use (&$generators)
+        {
+            if ($item instanceof \Generator) {
```

Comment:
shouldnt `instanceof \Closure` be generally supported? assuming it returns a generator

Reply:
> because the Closures are already called

well no :) https://3v4l.org/aDoJC

Reply:
@ro0NL your use case here is solve with `'lazy' => $heavy` instead

Reply:
@stof  you are right, i was to focused on a specific syntax. Do you think there's value in supporting closures, in ternms of lazy construction?

Reply:
> shouldnt instanceof \Closure be generally supported? assuming it returns a traversable

It should not be supported as in my point of view and because it would open a trap for application e.g.: You are doing loading data from the database. If you do it as a closure the method is called after the headers are already send means the response has a `Status Code 200`. If the closure will then throw a `throw new NoConnectionException()` because it can not reach the database. You can not change the Status Code and not even Symfony can create a 500 error then. That is why I would only support `\Generators`/`\Traversable` to avoid that trap. With only supporting \Generators the loading already started happening correctly.

> well no :) https://3v4l.org/aDoJC

Should be https://3v4l.org/RUKJB ;)

Reply:
I agree with @alexander-schranz here